components/automate-gateway/gateway/middleware/auth_interceptor_internal_test.go

```diff
@@ -74,6 +111,82 @@ func TestHeaderAuthValidForClientAndPeer(t *testing.T) {
 	}
 }
 
+func TestComboAuth(t *testing.T) {
+	albCert, _ := devCertToEncodedAndPeer(t, "automate-load-balancer")
+	agCert, agPeer := devCertToEncodedAndPeer(t, "automate-gateway")
+	otherServiceCert, _ := devCertToEncodedAndPeer(t, "deployment-service")
+
+	reqFromALBCtx := metadata.NewIncomingContext(
+		peer.NewContext(context.Background(), agPeer),
+		metadata.Pairs("x-client-cert", albCert, "grpcgateway-x-client-cert", agCert,
+			runtime.MetadataPrefix+"projects", "project1,project2",
+		))
+
+	reqDirectToGatewayCtx := metadata.NewIncomingContext(
+		peer.NewContext(context.Background(), agPeer),
+		metadata.Pairs("x-client-cert", albCert, "grpcgateway-x-client-cert", otherServiceCert,
+			runtime.MetadataPrefix+"projects", "project1,project2",
+		))
+
+	t.Run("It returns error for authn failure", func(t *testing.T) {
+		a := authInterceptor{
+			authn: testAuthnClient{err: errors.New("sample authn failure")},
+			authz: testAuthzHandler{},
+		}
+		_, err := a.combinedAuth(reqFromALBCtx, nil)
+		require.Error(t, err)
+	})
+	t.Run("It returns error for authz failure", func(t *testing.T) {
+		a := authInterceptor{
+			authn: testAuthnClient{},
+			authz: testAuthzHandler{err: errors.New("sample authz failure")},
+		}
+		_, err := a.combinedAuth(reqFromALBCtx, nil)
+		require.Error(t, err)
```

Comment:
this could assert that the error is [grpc code 7 Permission Denied](https://github.com/grpc/grpc-go/blob/master/codes/codes.go#L72)

Reply:
Same deal as above, this error is whatever the mocks provide.